pkg/infra/server.go

```diff
@@ -39,25 +45,29 @@ func NewServer(opts ...InitOption) (*Server, error) {
 
 	return &Server{
 		Config: cfg,
+
+		logger: logger,
 	}, nil
 }
 
 // ListenAndServe starts the JSON-RPC server
 func (s *Server) ListenAndServe() error {
-	rpcServer := server.NewRPCHandler()
+	rpcServer := server.NewRPCHandler(logging.Child(s.logger, "rpc_server"))
 
 	mux := http.NewServeMux()
 	rpcServer.Register(mux)
 
+	port := s.Config.HTTPConfig.Port
 	httpServer := &http.Server{
-		Addr:              fmt.Sprintf(":%d", s.Config.HTTPConfig.Port),
+		Addr:              fmt.Sprintf(":%d", port),
 		Handler:           mux,
 		ReadHeaderTimeout: 3 * time.Second,
 		ReadTimeout:       10 * time.Second,
 		WriteTimeout:      10 * time.Second,
 		IdleTimeout:       30 * time.Second,
 	}
 
+	logging.Sprintf(s.logger, slog.LevelInfo, "Listening on :%d", port)
```

Comment:
I'm not convinced that it's much more readable then
```suggestion
	s.logger.Info(fmt.Sprintf("http server started on port: %d", port))
```

Reply:
I've removed this function